Realm/Swift/RLMSupport.swift

```diff
@@ -29,18 +29,29 @@ extension RLMObject {
     }
 }
 
+public class RLMGenerator: GeneratorType {
+    private var i: UInt = 0
+    private let collection: RLMCollection
+
+    init(collection: RLMCollection) {
+        self.collection = collection
+    }
+
+    typealias Element = RLMObject
+
+    public func next() -> RLMObject? {
+        if i >= collection.count {
+            return .None
+        } else {
+            return collection[i++] as? RLMObject
+        }
+    }
+}
+
 extension RLMArray: SequenceType {
     // Support Sequence-style enumeration
     public func generate() -> GeneratorOf<RLMObject> {
-        var i: UInt  = 0
-
-        return GeneratorOf<RLMObject> {
-            if (i >= self.count) {
-                return .None
-            } else {
-                return self[i++] as? RLMObject
-            }
-        }
+        return GeneratorOf(RLMGenerator(collection: self))
```

Comment:
GeneratorOf went away in Swift 2.0, and there doesn't appear to be any equivalent generator from a block in the standard library, so it needs an explicit generator type (which turns out to be less code anyway...). Returning the type-erased generator isn't really necessary; it's just much more natural in 2.0 because it's a protocol (AnyGenerator) rather than a struct wrapping the inner generator.